### PR TITLE
Update to LESS 2.4.0

### PR DESCRIPTION
**Pull request by RobLoach**

### CODE REVIEW

File: src/less.plugin.coffee [hunk 1]
```diff
@@ -38,21 +38,10 @@ module.exports = (BasePlugin) ->
...
-
-					# Prepare
-					compileOptions =
-						compress: config.compress
-
-					# Extend Compile Options
-					compileOptions[key] = value  for own key,value of config.compileOptions  if config.compileOptions
-
-					# Compile
-					opts.content = tree.toCSS(compileOptions)
```
Comment (RobLoach) [inline src/less.plugin.coffee]: Not sure why we need this tree.toCSS.

File: src/less.plugin.coffee [hunk 2]
```diff
@@ -38,21 +38,10 @@ module.exports = (BasePlugin) ->
 				file.setMetaDefaults('referencesOthers': true)  if opts.content.indexOf('@import') isnt -1
 
 				# Parse
-				new (less.Parser)(parseOptions).parse opts.content, (err,tree) ->
-					# Check
+				less.render opts.content, parseOptions, (err, output) ->
 					if err
-						err = new Error(less.formatError(err, parseOptions))
 						return next(err)
```
Comment (RobLoach) [inline src/less.plugin.coffee]: It already gives us an err. No reason to build another one.

**Comment by balupton** [comment]
Are the comments here because of less API changes?

**Comment by RobLoach** [comment]
@balupton The comments I made? Was reviewing my own code.

**Comment by patrixd** [comment]
Please I need this update. The current less version is really old and I can't use calc properly without strict-math configuration option 'on'

**Comment by derberg** [comment]
Hi guys, any news about this pull request? why it is not merged yet? these errors on travis do not provide any explanation

File: src/less.plugin.coffee (hunk 1 above)
Comment (balupton) [inline src/less.plugin.coffee]: Was this change needed? It may be necessary to enable/disable compression

File: src/less.plugin.coffee (hunk 2 above)
Comment (balupton) [inline src/less.plugin.coffee]: I believe this makes the error a lot more readable, such as including the line numbers of input source and sometimes a code sample

**Comment by balupton** [comment]
@derberg posted up some follow up questions for @RobLoach 

Ideally, I'd have this below feature from https://bevry.me/patreon done before working on more plugins:

```
Abstract out the powerful DocPad Plugin System into its own Independent Plugin System.
This means taking the powerful and time-tested DocPad Plugin architecture, and abstracting it out into an independent system that can be used for any project to implement it's own plugin system easily. Existing functionality allows plugins to listen to events that can even be asynchronous which are executed safely an in prioritised order either in parallel or serially by the mediator's choosing. New features will include semantic versioning support for plugins, as well as automatic installation of the correct plugin version for your existing peer dependencies.
```

**Comment by balupton** [comment]
added in the now released v2.5.0
